include/networkit/scd/CliqueDetect.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef NETWORKIT_SCD_CLIQUE_DETECT_HPP_
+#define NETWORKIT_SCD_CLIQUE_DETECT_HPP_
+
+#include <networkit/scd/SelectiveCommunityDetector.hpp>
+
+namespace NetworKit {
+
+/**
+ * The CliqueDetect algorithm. First finds the largest clique in the seed node's
+ * neighborhood.
+ *
+ * The algorithm can handle weighted graphs.
+ */
+class CliqueDetect : public NetworKit::SelectiveCommunityDetector {
+
+public:
+    /**
+     * Construct a Cliquedetect object.
+     *
+     * @param[in] G The graph to detect communities on
+     */
+    CliqueDetect(const Graph &G);
+
+    /**
+     * Expands a single seed node/vertex into a maximal clique.
+     *
+     * @param[in] s the seed node
+     * @return A community of the seed node
+     */
+    std::set<node> expandOneCommunity(node seed) override;
+
+    /**
+     * Detect a single clique for the given seed nodes.
+     *
+     * The resulting community is a clique iff the seeds form a clique.
+     * Otherwise, only the added nodes form a clique that is fully connected
+     * to the seed nodes.
+     *
+     * @param seeds The seeds for the community.
+     * @return The found community as set of nodes.
+     */
+    std::set<node> expandOneCommunity(const std::set<node> &seeds) override;
+
+protected:
+    template <typename F>
+    std::vector<node> getMaximumWeightClique(const std::vector<node> &nodes, F is_seed) const;
+
+    Graph createSubgraphFromNodes(const std::vector<node> &nodes) const;
```

Comment:
Isn't this method already implemented in `GraphTools::subgraphFromNodes`?

Reply:
No, the method in `GraphTools` is linear in the size of the input graph and creates a graph with the same node ids as the input graph (i.e., O(n) nodes), while this method creates a graph with compact node ids and is linear in the volume of the given nodes. It could make sense to offer this method with compact node ids in `GraphTools`, though.

Reply:
In that case I'd suggest to move this method to `GraphTools` and provide some unit tests.